scopes/component/ui/component-artifacts/artifacts-tree/artifacts-tree.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { HTMLAttributes, useMemo, useState, useContext } from 'react';
+import classNames from 'classnames';
+import { Icon } from '@teambit/evangelist.elements.icon';
+import { WidgetProps, TreeNode as Node } from '@teambit/ui-foundation.ui.tree.tree-node';
+import { DrawerUI } from '@teambit/ui-foundation.ui.tree.drawer';
+import { FileTree, useFileTreeContext } from '@teambit/ui-foundation.ui.tree.file-tree';
+import { useComponentArtifacts } from '@teambit/component.ui.artifacts.queries.use-component-artifacts';
+import { useLocation } from '@teambit/base-react.navigation.link';
+import { ArtifactFile } from '@teambit/component.ui.artifacts.models.component-artifacts-model';
+import { TreeNode, TreeNodeProps } from '@teambit/design.ui.tree';
+import { TreeContext } from '@teambit/base-ui.graph.tree.tree-context';
+import { ComponentTreeLoader } from '@teambit/design.ui.skeletons.sidebar-loader';
+import { ComponentID } from '@teambit/component-id';
+import { FolderTreeNode } from '@teambit/ui-foundation.ui.tree.folder-tree-node';
+
+import styles from './artifacts-tree.module.scss';
+
+export type ArtifactsTreeProps = {
+  getIcon?: (node: TreeNode) => string | undefined;
+  drawerOpen: boolean;
+  onToggleDrawer: () => void;
+  host: string;
+  componentId: ComponentID;
+  drawerName: string;
+} & HTMLAttributes<HTMLDivElement>;
+
+const fileNodeClicked = (files: (ArtifactFile & { id: string })[], opts: 'download' | 'new tab') => (e, node) => {
```

Comment:
lets move all except from `ArtifactsTree` to their own files. would clean this up a bit